### DISCUSSION TITLE
Execute RegisterImplementationSourceOutput last?

### DISCUSSION THREAD

**Discussion by weifenluo** [dotnet/roslyn#73888] [Q&A] [answered]
When will source generator with `RegisterImplementationSourceOutput` get executed? Can I get it executed after other source generator(s)? Came across with #57963 and #54723, I still can't get a clear picture.

For simplicity, let's say I have two incremental source generators: `KeyGenerator` and `TypeDescriptorGenerator`. Given the following input source code:
```
public partial class MyEntityType : IEntityType
{
    public Field<int> Id { get; private set; }

    public Field<string> Name { get; private set; }

    [GenerateKey(nameof(Id))]
    public partial class Key : IEntityType
    {
    }
}
```

The `KeyGenerator` will take the `[GenerateKey]` marker attribute to generate the following code, based on the field name(s) provided:
```
// <AutoGenerated/>
partial class Key
{
    // copy from containing MyEntityType:
    public Field<int> Id { get; private set; }
}
```

The `TypeDescriptorGenerator`, on the other hand, will take all types which implements `IEntityType`, taking all the fields of the type, to generate a type descriptor class. The generated type descriptor class is invisible to developer, so it's registered with `RegisterImplementationSourceOutput`. The generation should take the output `Compilation` of `KeyGenerator`, because it should take the generated `Key.Id` property into account; this code generation is expensive, for performance consideration, ideally it should only run once.

So back to the question: can I have `TypeDescriptorGenerator` only run at the end of compile pipeline? If yes, how? If no, what's the workaround?

**Comment by Dreamescaper** [comment, marked as answer]
Source Generators do not affect each other at all:
https://github.com/dotnet/roslyn/issues/57239
